greengrassTestSuite.py

```diff
@@ -12,8 +12,43 @@
 
 
 class SystemInterface:
-    def __init__(self):
-        pass
+
+    def check_systemctl_status_for_component(self, component_name):
+        try:
+            cmd = [
+                "sudo",
+                "systemctl",
+                "status",
+                f"ggl.{component_name[0]}.service",
+            ]
+
+            # Run the command and stream output
+            process = subprocess.Popen(
+                cmd,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                universal_newlines=True,
+            )
+
+            output = process.stdout.read()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The use of `subprocess.Popen.stdout.read()` might deadlock if any OS pipe buffers fill up and block the child process. We recommend using the `communicate()` API.

[Learn more](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate)